Modules/Retailer/Http/Services/RetailerService.php

```diff
@@ -0,0 +1,256 @@
+        $city = $request->citySelect;
+        $retailers = User::where('role', '=', ROLE::ROLE_RETAILER)
+            ->whereIn('id', function ($query) {
+                $query->select('user_id')
+                    ->from('retailers');
+            })->with('retailer');
+
+        if (!empty($country)) {
+            $retailers->whereHas('retailer', function ($query) use ($country) {
+
+                $query->where('country', $country);
+            });
+        }
+        if (!empty($state)) {
+            $retailers->whereHas('retailer', function ($query) use ($state) {
+                $query->where('state', $state);
+            });
+        }
+        if (!empty($city)) {
+            $retailers->whereHas('retailer', function ($query) use ($city) {
+
+                $query->where('town', $city);
+            });
+        }
+
+        if (!empty($catID)) {
+            $retailers->whereHas('cart', function ($query) use ($catID) {
+                $query->whereHas('product', function ($query) use ($catID) {
+                    $query->where('main_category', $catID);
+                });
+            })->with(['cart.product', 'retailer' => function ($query) {
+                $query->select('user_id');
+            }])->with('retailer');
+        }
+        $filteredCount = $retailers->count();
+        $start = $request->start ?? 0;
+        $length = $request->length ?? 0;
+        $retailers = $retailers->offset($start)->limit($length);
+        $result = DataTables::of($retailers)
+            ->addIndexColumn()
+            ->addColumn('store_name', function ($row) {
+                return $row->retailer->store_name;
+            })
+            ->addColumn('store_type', function ($row) {
+                return $row->retailer->store_type;
+            })
+            ->addColumn('action', function ($row) {
+                return '<a href="' . url('/retailer/details/' . $row->id) . '"><span class="fas fa-eye"></span></a>  <a href="' . url('/order/retailer-order/' . $row->id) . '"><span class="fas fa-file-invoice-dollar"></span></a>  <a href="javascript:void(0)" onclick="event.preventDefault(); confirmDelete(' . $row->id . ');"><span class="fas fa-trash"></span></a> ';
+            })
+            ->rawColumns(['action'])
+            ->make(true);
+        $resultData = $result->getData(true);
+        $resultData['recordsFiltered'] = $filteredCount;
+        $resultData['draw'] = $request->draw;
+
+        return $resultData;
+    }
+
+    /**
+     * Details of retailer
+     *
+     * @param $id
+     * @return mixed
+     */
+    public function detail($id): mixed
+    {
+        $userRetailer = User::with('retailer')->findOrFail($id);
+        $detail = $userRetailer->retailer;
+        $country = Country::where('id', $userRetailer->retailer->country)->first();
+        $state = State::where('id', $userRetailer->retailer->state)->first();
+        $town = City::where('id', $userRetailer->retailer->town)->first();
```

Comment:
Make relationship b/w Countries, States and Cities.
Then when you get the country, get the relational data in that single query instead of getting in 3 different queries.